### PR TITLE
Adds R4B

### PR DESCRIPTION
**Pull request by brendankowitz**
## Description
- Adds R4B projects 🚀 
- Some tweaks to keep build times down (zipping e2e tests). Config to allow unit test assemblies to run in parallel 🚅 
- Splits hosting plan per FHIR version in PR/CI builds, the extra 2 sites seemed to be overloading the B3 instance

## Related issues
Addresses AB#95729

## Testing
- Adds new R4B test projects
- Some manual testing with .http scripts

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with Azure API for FHIR if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [x] Tag the PR with Azure Healthcare APIs if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [ ] [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) CI is green before merge 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Feature

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Microsoft.Health.Fhir.Core.csproj [hunk 1]
```diff
@@ -29,6 +29,16 @@
     <EmbeddedResource Include="Data\Stu3\unsupported-search-parameters.json" />
     <EmbeddedResource Include="Data\Stu3\resourcepath-codesystem-mappings.json" />
     <EmbeddedResource Include="Features\Security\roles.schema.json" />
+    <None Remove="Data\R4B\BaseCapabilities.json" />
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Core/Microsoft.Health.Fhir.Core.csproj:32]: Are the None lines needed here? They aren't present for any of the other embedded resources.

File: src/Microsoft.Health.Fhir.R4B.Api/Features/Resources/Bundle/BundleHandler.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Fhir.Api.Features.Resources.Bundle
+{
+    /// <summary>
+    /// FHIR R4 specific portions of the bundle handler.
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.R4B.Api/Features/Resources/Bundle/BundleHandler.cs:9]: Nit: R4B in the comment

File: src/Microsoft.Health.Fhir.R4B.Core/VersionSpecificModelInfoProvider.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core
+{
+    /// <summary>
+    /// Provides access to R4 FHIR Models and Resources
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.R4B.Core/VersionSpecificModelInfoProvider.cs:11]: Nit: R4B in comment

**Review by LTA-Thinking — APPROVED** [review]

File: build/ci-pipeline.yml [hunk 4]
```diff
@@ -170,6 +194,19 @@ stages:
...
+  dependsOn:
+  - BuildUnitTests
+  - redeployR4B
+  - redeployR4BSql
+  jobs:
+  - template: ./jobs/run-tests.yml
+    parameters:
+      version: R4B
+      keyVaultName: $(DeploymentEnvironmentNameR4)
+      appServiceName: $(DeploymentEnvironmentNameR4)
```
Comment (brendankowitz) [inline build/ci-pipeline.yml:208]:
```suggestion
      keyVaultName: $(DeploymentEnvironmentNameR4B)
      appServiceName: $(DeploymentEnvironmentNameR4B)
```

**Review by LTA-Thinking — DISMISSED** [review]

**Review by LTA-Thinking — DISMISSED** [review]

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**